codejail/safe_exec.py

```diff
@@ -4,38 +4,20 @@
 import os.path
 import shutil
 import sys
-import textwrap
 
-try:
-    import simplejson as json
-except ImportError:
-    import json
-
-from codejail import jail_code
-from codejail.util import temp_directory, change_directory
+from .exceptions import SafeExecException
+from . import jail_code
+from .util import temp_directory, change_directory, json_safe
 
 log = logging.getLogger("codejail")
 
 
 # Flags to let developers temporarily change some behavior in this file.
 
-# Set this to True to log all the code and globals being executed.
-LOG_ALL_CODE = False
 # Set this to True to use the unsafe code, so that you can debug it.
 ALWAYS_BE_UNSAFE = False
 
 
-class SafeExecException(Exception):
-    """
-    Python code running in the sandbox has failed.
-
-    The message will be the stdout of the sandboxed process, which will usually
-    contain the original exception message.
-
-    """
-    pass
-
-
 def safe_exec(code, globals_dict, files=None, python_path=None, slug=None,
```

Comment:
The reason for moving the code to the jail objects was (primarily) to make it easier to access the language specific configuration, especially for safe_exec (which relies specifically on a jail named "python".  The reason for leaving the functional version is to maintain backwardscompatibility for existing code that uses it.  I agree that the duplication of docstrings is unfortunate.  I could remove the bulk of the documentation from the functional versions and replace it with a note on how the Jail methods are invoked, with a cross-reference to the Jail documentation.